lib/elixir/lib/keyword.ex

```diff
@@ -69,9 +69,7 @@ defmodule Keyword do
   """
   @spec new(Enum.t) :: t
   def new(pairs) do
-    Enum.reduce pairs, [], fn {k, v}, keywords ->
-      put(keywords, k, v)
-    end
+    Enum.uniq_by(Enum.reverse(pairs), fn {x, _} when is_atom(x) -> x end)
```

Comment:
The timings are following:
```elixir
new:    1000000   1.44 µs/op
old:    1000000   2.03 µs/op

new:     500000   6.09 µs/op
old:     500000   7.86 µs/op
```
It seems that previous implementation of `Enum.uniq_by/2` (https://github.com/elixir-lang/elixir/commit/97365935835c703b71954bd8fad7240fbaaf962f) was more performant:
```elixir
new:   10000000   0.58 µs/op
old:    1000000   1.99 µs/op

new:    1000000   2.65 µs/op
old:     500000   7.88 µs/op
```